os/drivers/audio/audio_null.c

```diff
@@ -490,6 +612,95 @@ static void *null_workerthread(pthread_addr_t pvarg)
 	return NULL;
 }
 
+/****************************************************************************
+ * Name: null_unregisterprocess
+ *
+ * Description: Register the process (epd, wakeup etc.).
+ *
+ ****************************************************************************/
+static int null_unregisterprocess(FAR struct audio_lowerhalf_s *dev)
+{
+	FAR struct null_dev_s *priv = (FAR struct null_dev_s *)dev;
+	void *value;
+	int ret;
+	/* Join any old worker thread we had created to prevent a memory leak */
+	ret = OK;
+	if (priv->worker_threadid != 0) {
+		ret = pthread_join(priv->process_threadid, &value);
+		if (ret != OK) {
+			auddbg("join failed!! ret : %d\n", ret);
+			return ret;
+		}
+		priv->worker_threadid = 0;
+	}
+
+	if (priv->dev.process_mq != NULL) {
+		mq_close(priv->dev.process_mq);
+		priv->dev.process_mq = NULL;
+	} else {
+		auddbg("mq is null\n");
+		return -ENOENT;
+	}
+#ifdef CONFIG_AUDIO_SPEECH_DETECT_FEATURES
+	priv->speech_state = AUDIO_NULL_SPEECH_STATE_NONE;
+#ifdef CONFIG_AUDIO_KEYWORD_DETECT
+	priv->keyword_detect = false;
+#endif
+#ifdef CONFIG_AUDIO_ENDPOINT_DETECT
+	priv->endpoint_detect = false;
+#endif
+
+#endif
+	return ret;
+}
+
+/****************************************************************************
+ * Name: null_processstart
+ *
+ * Description:
+ *   Register the process (epd, wakeup etc.).
+ *
+ ****************************************************************************/
+static int null_registerprocess(FAR struct audio_lowerhalf_s *dev, mqd_t mq)
+{
+	FAR struct null_dev_s *priv = (FAR struct null_dev_s *)dev;
+	pthread_attr_t tattr;
+	FAR void *value;
+	int ret;
+
+	audvdbg("Entry\n");
+	if (priv->dev.process_mq == NULL) {
+		priv->dev.process_mq = mq;
+	} else {
+		auddbg("already registered!!\n");
+		return -EBUSY;
+	}
+
+	/* Join any old worker thread we had created to prevent a memory leak */
+
+	if (priv->process_threadid != 0) {
+		audvdbg("Joining old thread\n");
```

Comment:
At this line, process_terminate isn't set as false. Is the null_processthread finishing itself automatically?

Reply:
Null driver is kind of sample, it will change value to 'true' when user calls ioctl with AUDIOIOC_STOPPROCESS.
Also will be destroyed by AUDIOIOC_UNREGISTERPROCESS.
(Initialize : false)
register : false -> start -> stop : true -> unregister